datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -738,6 +740,8 @@ fn temporal_coercion(lhs_type: &DataType, rhs_type: &DataType) -> Option<DataTyp
         (Interval(YearMonth), Interval(YearMonth)) => Some(Interval(YearMonth)),
         (Interval(DayTime), Interval(DayTime)) => Some(Interval(DayTime)),
         (Interval(_), Interval(_)) => Some(Interval(MonthDayNano)),
+        (Date32, Date32) => Some(Date32),
```

Comment:
Maybe we should put a generic clause that if `lhs_type == rhs_type` we can return that type?

Reply:
Yes!